GitHub.Authentication/Authority.cs

```diff
@@ -177,6 +177,20 @@ public Authority(TargetUri targetUri)
                                     return new AuthenticationResult(GitHubAuthenticationResultType.Failure);
                                 }
                             }
+                        case HttpStatusCode.Forbidden:
+                            // This API only supports Basic authentication. If a valid OAuth token is supplied
+                            // as the password, then a Forbidden response is returned instead of an Unauthorized.
+                            // In that case, the supplied password is an OAuth token and is valid and we don't need
+                            // to create a new personal access token.
+                            var contentBody = await response.Content.ReadAsStringAsync();
+                            if (contentBody.Contains("This API can only be accessed with username and password Basic Auth"))
```

Comment:
So this is the "magic".

Any chance github.com is really smart and localized based on browser settings? If so, does the string remain in English?

Reply:
github.com content (and especially the API) is not localized on browser settings.